packages/nextjs/src/server/clerkMiddleware.ts

```diff
@@ -465,7 +449,7 @@ const handleControlFlowErrors = (
   requestState: RequestState,
 ): Response => {
   if (isNextjsUnauthorizedError(e)) {
-    const response = NextResponse.next({ status: 401 });
+    const response = new NextResponse(null, { status: 401 });
```

Comment:
We need to terminate requests for invalid machine tokens when used inside middleware. The previous one continues processing the middleware chain instead of ending the request.